test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1112 @@
+
+                self.server.expect(RESV, attrs, id=r)
+                acct_msg = "Y;" + r + ";requestor=Scheduler@"
+                acct_msg += self.server.shortname + " start="
+                acct_msg += str(new_start) + " end=" + str(new_end)
+                self.server.status(RESV, 'resv_nodes', id=r)
+                acct_msg += " nodes="
+                acct_msg += re.escape(self.server.reservations[r].resvnodes())
+
+                if r[0] == 'S':
+                    self.server.status(RESV, 'reserve_count', id=r)
+                    count = self.server.reservations[r].attributes[
+                        'reserve_count']
+                    acct_msg += " count=" + count
+
+                self.server.accounting_match(acct_msg, regexp=True, interval=2,
+                                             max_attempts=30)
+
+                # Check if reservation reports new start time
+                # and updated duration.
+
+                msg = "Resv;" + r + ";Reservation alter confirmed"
+            else:
+                msg = "Resv;" + r + ";Reservation alter denied"
+
+            lines = self.server.log_match(msg, n='ALL', allmatch=True,
+                                          interval=2, max_attempts=30)
+            self.assertEqual(len(lines), sequence)
+
+            return new_start, new_end
+        else:
+            try:
+                self.server.alterresv(r, attrs)
+            except PbsResvAlterError as e:
+                if e.rc == self.PBSE_RESV_NOT_EMPTY:
+                    msg = "pbs_ralter: Reservation not empty"
+                elif e.rc == self.PBSE_UNKRESVID:
+                    msg = "pbs_ralter: Unknown Reservation Id"
+                elif e.rc == self.PBSE_STDG_RESV_OCCR_CONFLICT:
+                    msg = "Requested time(s) will interfere with "
+                    msg += "a later occurrence"
+                    log_msg = "Resv;" + r + ";" + msg
+                    msg = "pbs_ralter: " + msg
+                    self.server.log_match(log_msg, interval=2,
+                                          max_attempts=30)
+                elif e.rc == self.PBSE_INCORRECT_USAGE:
+                    pass
+
+                self.assertNotEqual(e.msg, msg)
+                return start, end
+            else:
+                self.assertFalse("Reservation alter allowed when it should" +
+                                 "not be.")
+
+    def test_alter_advance_resv_start_time_before_run(self):
+        """
+        This test case covers the below scenarios for an advance reservation
+        that has not started running.
+
+        1. Make an advance reservation start late (empty reservation)
+        2. Make an advance reservation start early (empty reservation)
+        3. Make an advance reservation start early (with a job in it)
+
+        All the above operations are expected to be successful.
+        """
+        offset = 120
+        duration = 120
+        shift = 60
```

Comment:
These values are way too high for automation purposes.
The test suite will consume a lot of time to complete. Please optimize them to the best.
This comment applies for all the test cases. Please take a look and optimize the time whatever best can be done.

Reply:
Hmmm. This can be done. I have reduced the duration and offsets to reduce the run time of the test suits.